File: app/src/main/java/mobi/moop/features/chamado/ChamadoViewHolder.java
package mobi.moop.features.chamado;

import android.content.Context;
import android.graphics.Color;
import android.graphics.ColorFilter;
import android.graphics.PorterDuff;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.text.SimpleDateFormat;

import butterknife.BindView;
import butterknife.ButterKnife;
import mobi.moop.R;
import mobi.moop.model.entities.Chamado;
import mobi.moop.model.entities.StatusChamado;

/**
 * Created by murilo aires on 02/10/2017.
 */

public class ChamadoViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.text_titulo_chamado)
    TextView textTituloChamado;

    @BindView(R.id.text_descricao)
    TextView textDescricaoChamado;

    @BindView(R.id.status_background)
    View statusBackground;

    @BindView(R.id.textoStatus)
    TextView textStatus;

    @BindView(R.id.text_data_abertura)
    TextView textDataAberturaChamado;

    @BindView(R.id.img_foto_chamado)
    ImageView imgChamado;

    private final Context context;
    private Chamado chamado;

    public ChamadoViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        this.context = itemView.getContext();
    }

    public void bindView(Chamado chamado) {
        this.chamado = chamado;
        populateFields();
    }

    private void populateFields() {
        switch (chamado.getStatus()) {
            case StatusChamado.ABERTO:
                textStatus.setText(context.getString(R.string.aberto));
                statusBackground.getBackground().setColorFilter(context.getResources().getColor(android.R.color.holo_red_dark), PorterDuff.Mode.MULTIPLY);
                break;
            case StatusChamado.EM_ANDAMENTO:
                textStatus.setText(context.getString(R.string.em_andamento));
                statusBackground.getBackground().setColorFilter(context.getResources().getColor(R.color.amarelo_analise), PorterDuff.Mode.MULTIPLY);
                break;
            case StatusChamado.RESOLVIDO:
                textStatus.setText(context.getString(R.string.resolvido));
                statusBackground.getBackground().setColorFilter(Color.GREEN, PorterDuff.Mode.MULTIPLY);
                break;
            default:
                textStatus.setText(context.getString(R.string.cancelado));
                statusBackground.getBackground().setColorFilter(context.getResources().getColor(android.R.color.darker_gray), PorterDuff.Mode.MULTIPLY);
        }
        textTituloChamado.setText(chamado.getTitulo());
        textDescricaoChamado.setText(chamado.getTexto());
//        Picasso.with(context).load(chamado).error()
        textDataAberturaChamado.setText("Aberto em " + new SimpleDateFormat("dd/MM/yyyy").format(this.chamado.getUpdatedAt()) + " às " + new SimpleDateFormat("HH:mm").format(chamado.getUpdatedAt()));

    }
}

File: app/src/main/java/brmobi/moop/ui/tickets/ChamadoViewHolder.kt
package brmobi.moop.ui.tickets

import android.content.Context
import android.graphics.Color
import android.graphics.PorterDuff
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.View
import brmobi.moop.R
import brmobi.moop.data.network.model.Chamado
import brmobi.moop.data.network.model.StatusChamado
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.item_chamado.view.*
import java.text.SimpleDateFormat

/**
 * Created by murilo aires on 02/10/2017.
 */

class ChamadoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    val textTicketTitle = itemView.text_titulo_chamado
    val textTicketDescription = itemView.text_descricao
    val statusBackground = itemView.status_background
    val textStatus = itemView.textoStatus
    val textOpenningTime = itemView.text_data_abertura
    val imgTicket = itemView.img_foto_chamado

    private val context: Context = itemView.context


    fun bindView(chamado: Chamado) {
        populateFields(chamado)
    }

    private fun populateFields(chamado: Chamado) {
        when (chamado!!.status) {
            StatusChamado.ABERTO -> {
                textStatus.text = context.getString(R.string.aberto)
                statusBackground!!.background.setColorFilter(context.resources.getColor(android.R.color.holo_red_dark), PorterDuff.Mode.MULTIPLY)
            }
            StatusChamado.EM_ANDAMENTO -> {
                textStatus.text = context.getString(R.string.em_andamento)
                statusBackground!!.background.setColorFilter(context.resources.getColor(R.color.amarelo_analise), PorterDuff.Mode.MULTIPLY)
            }
            StatusChamado.RESOLVIDO -> {
                textStatus.text = context.getString(R.string.resolvido)
                statusBackground.background.setColorFilter(Color.GREEN, PorterDuff.Mode.MULTIPLY)
            }
            else -> {
                textStatus.text = context.getString(R.string.cancelado)
                statusBackground.background.setColorFilter(context.resources.getColor(android.R.color.darker_gray), PorterDuff.Mode.MULTIPLY)
            }
        }
        textTicketTitle.text = chamado!!.titulo
        textTicketDescription.text = chamado!!.texto

        textOpenningTime.text = "Aberto em " + SimpleDateFormat("dd/MM/yyyy").format(chamado.updatedAt) + " às " + SimpleDateFormat("HH:mm").format(chamado!!.updatedAt)
        if (!chamado.imagem.isNullOrEmpty()) {
            Picasso.with(context).load(chamado.imagem).into(imgTicket)
        } else {
            imgTicket.setColorFilter(ContextCompat.getColor(context, R.color.cinza))
        }

    }
}
